js/main.js

```diff
@@ -5333,11 +5338,11 @@ function insertTableRows(table, rows) {
 	$(rows).each(function() {
         var row = $(this);
         // By default workshops stay hidden
-        row_html += `<tr class="${row[2] == 'schedule-workshop'? row[2] + ' table-row-hidden' : row[2]}">
+        row_html += `<tr class="${row[2] == 'schedule-workshop' ? row[2] + ' table-row-hidden' : row[2]}">
                        <td class="first_col">${row[0]}</td>
-                       <td class="second_col" id=talkno${((row[1]!='') ? row[3] || '-1' : -1)}>${row[1]}</td>
+                       <td class="second_col" id=talkno${row[1] != '' ? row[2] == 'schedule-keynote' ? row[3] || - 1 : -1 : -1}>${row[1]}</td>
```

Comment:
```suggestion
                       <td class="second_col" id=talkno${row[1] != '' ? row[2] : -1}>${row[1]}</td>
```

Removing this makes the modal work for talks, workshops and keynotes.